src/main/java/de/codecentric/centerdevice/util/StageUtils.java

```diff
@@ -46,12 +56,49 @@ public static void focusNextStage() {
 		}
 	}
 
+	@SuppressWarnings("unchecked")
 	public static ObservableList<Stage> getStages() {
-		return StageHelper.getStages();
+		if (stages == null) {
+			// Java 9
+			try {
+				windows = (ObservableList<Window>)Window.class.getMethod("getWindows").invoke(null);
+				stages = FXCollections.observableArrayList();
+				windows.addListener(new ListChangeListener<Window>() {
+					@Override
+					public void onChanged(javafx.collections.ListChangeListener.Change<? extends Window> c) {
+						updateStages();
```

Comment:
I think we also need to take care of windows being removed. I think right now, on deletions and permutations of `windows` we'll re-add all stages to `stages` in `updateStages`. 
I think we'd need something like here: https://docs.oracle.com/javase/8/javafx/api/javafx/collections/ListChangeListener.Change.html

    public void onChanged(Change<tem> c) {
         while (c.next()) {
             if (c.wasPermutated()) {
                     // no one cares
                 } else if (c.wasUpdated()) {
                     // no one cares
                 } else {
                     for (Item remitem : c.getRemoved()) {
                         // Remove stage
                     }
                     for (Item additem : c.getAddedSubList()) {
                         // add stage
                     }
                 }
             }
         }
     });

What do you think?